image/manifest.go

```diff
@@ -238,5 +247,27 @@ loop:
 			return errors.Wrap(err, "error changing time")
 		}
 	}
+
+	return nil
+}
+
+func applyOwnership(path string, hdr *tar.Header) error {
+	switch hdr.Typeflag {
+	case tar.TypeSymlink:
+		if err := os.Lchown(path, hdr.Uid, hdr.Gid); err != nil {
+			return errors.Wrap(err, "error chowing symlink")
+		}
+
+	case tar.TypeReg, tar.TypeRegA:
+		if err := os.Chown(path, hdr.Uid, hdr.Gid); err != nil {
+			return errors.Wrap(err, "error chowing file")
+		}
+
+	case tar.TypeDir:
+		if err := os.Chown(path, hdr.Uid, hdr.Gid); err != nil {
+			return errors.Wrap(err, "error chowing directory")
```

Comment:
Agreed, I am not a big fun of it either. I would even go one step further into dropping the switch altogether§ and calling `Lchown` for every entry (file, directory or symlink).